### ISSUE TITLE
[BUG]: when add install tpot, then pycaret compare_models MAE increased

### ISSUE THREAD

**Issue by showkeyjar** [pycaret/pycaret#3983]
### pycaret version checks

- [X] I have checked that this issue has not already been reported [here](https://github.com/pycaret/pycaret/issues).

- [X] I have confirmed this bug exists on the [latest version](https://github.com/pycaret/pycaret/releases) of pycaret.

- [ ] I have confirmed this bug exists on the master branch of pycaret (pip install -U git+https://github.com/pycaret/pycaret.git@master).

### Issue Description

I installed pycaret in conda env, and try to compare models with my training data for a regression task,
but it had different results every time, even set session_id and random seed.

so, I continue installed tpot to check result, but when I ran pycaret again, the MAE grew 5x than before.

I feel very confused, I'm running pycaret in jupyterlab, and the console has no error.

### Reproducible Example

```python
`pip install pycaret[full]`

from pycaret.regression import RegressionExperiment

exp.setup(df_samples[ml_sel_cols], target='y', session_id=123)

best = exp.compare_models(exclude=['lr', 'lar', 'llar', 'lasso', 'omp', 'knn', 'huber', 'par', 'ridge', 'dummy', 'catboost'])

`conda install -c conda-forge tpot`

then run upper code again
```

### Expected Behavior

the MAE of two times must be same or near

### Actual Results

```python-traceback
but the first best MAE is 0.98

and the second best MAE is 4.99
```

### Installed Versions

python 3.11
conda 22.90
pycaret 3.3.1 and 3.3.0 both tested
tpot 0.12.2
jupyterlab 4.1.5

**Comment by showkeyjar** [comment]
and I found new bugs, maybe is the same of above: when comparing_models multiple times and testing again, the pipline of the previous version of the model may be used.

it happend in jupyterlab, I used two different datas(data1, data2) to compare_models twice, but the second time predict raise error: no columns XXX found. those missing columns are in data1

it works when I saved model and restart kerenal and rerun it.

**Comment by celestinoxp** [comment]
@showkeyjar There may be some conflict in the packages and possible interference in the results...

do you have any solution? Can you create a pull-request?

**Comment by moezali1** [comment]
Closing as part of the 4.0 backlog cleanup — see #4177.

PyCaret 4.0 is a ground-up rewrite (OOP-only API, sklearn-native engine, Python 3.11+, sklearn ≥ 1.7) and almost all pre-4.0 issues either reference removed APIs (`setup()`, `plot_model`, `evaluate_model`, `interpret_model`, `automl`, the legacy time-series module, the Streamlit dashboard, …) or no longer reproduce on the new code path.

**If this still affects you in 4.0, please re-file as a new issue** with:

- the experiment constructor call (`Experiment(...)`),
- the operation that fails,
- the full traceback,
- your Python / sklearn / pycaret versions.

Resources for 4.0:

- Docs: https://pycaret.org/docs
- Migrate from 3.x: https://pycaret.org/docs/guides/migrating-from-3-x
- API reference: https://pycaret.org/reference
- FAQ: https://pycaret.org/docs/resources/faqs

Thanks for filing — and apologies for the long latency. The repo had drifted; we're starting clean.
